src/app/info/WikiItems.tsx

```diff
@@ -52,40 +49,83 @@ export const convertContentToElements = (rowContent: string): React.JSX.Element[
     });
 };
 
-const WikiItems: React.FC<WikiItemsProps> = (props) => {
-    const sortedRows: DbWikiRow[] = props.rows.slice().sort((r1: DbWikiRow, r2: DbWikiRow) => {
-        return r1.order > r2.order ? 1 : -1;
-    });
+const WikiItem: React.FC<WikiItemProps> = ({ row, appendNewRow, removeRow }) => {
     return (
-        <>
-            {sortedRows.map((row: DbWikiRow) => {
-                return <WikiItem key={row.order} row={row} />;
-            })}
-        </>
+        <WikiItemPositioner>
+            <EditModeDependentItem row={row} appendNewRow={appendNewRow} removeRow={removeRow} />
+        </WikiItemPositioner>
     );
 };
 
-const WikiItem: React.FC<WikiItemProps> = ({ row }) => {
+const WikiItems: React.FC<WikiItemsProps> = ({ rows }) => {
+    const sortedRows: DbWikiRow[] = rows.slice().sort((r1: DbWikiRow, r2: DbWikiRow) => {
+        return r1.row_order > r2.row_order ? 1 : -1;
+    });
+
+    const [displayRows, setDisplayRows] = React.useState<DbWikiRow[]>(sortedRows);
+
+    const wikiItemsEndRef = useRef<HTMLDivElement>(null);
+    useEffect(() => {
+        if (wikiItemsEndRef.current) {
+            wikiItemsEndRef.current.scrollIntoView({
+                behavior: "smooth",
+            });
+        }
+    }, [displayRows]);
+
+    const [doesEmptyRowExist, setDoesEmptyRowExist] = useState<boolean>(false);
+
+    useMemo(() => {
+        setDoesEmptyRowExist(
+            displayRows.filter((row) => {
+                return !row.title && !row.content;
+            }).length !== 0
+        );
+    }, [displayRows]);
```

Comment:
`useMemo` is for caching values, so we can completely avoid state here (I've also rewritten the function for a bit more conciseness):
```
    const doesEmptyRowExist = useMemo(
        () => displayRows.some((row) => !row.title && !row.content),
        [displayRows]
    );
```